metals/src/main/scala/scala/meta/internal/metals/Doctor.scala

```diff
@@ -204,37 +211,65 @@ final class Doctor(
 
   private def buildTargetsHtml(): String = {
     new HtmlBuilder()
-      .element("h1")(_.text("Metals Doctor"))
+      .element("h1")(_.text(doctorTitle))
       .call(buildTargetsTable)
       .render
   }
 
+  private def buildTargetsJson(): String = {
+    val json = ujson
+      .Obj(
+        "name" -> doctorTitle,
+        "headerText" -> doctorHeading
+      )
+
+    val targets = buildTargets.all.toList
+
+    if (targets.isEmpty) {
+      json("messages") = Arr(
+        ujson.Obj(
+          "title" -> noBuildTargetsTitle,
+          "recommendations" -> Arr(noBuildTargetRecOne, noBuildTargetRecTwo)
+        )
+      )
+      ujson.write(json)
+    } else {
+      val targetResults = targets.sortBy(_.info.getBaseDirectory).map {
+        target =>
+          val targetInfo = extractTargetInfo(target)
+          ujson.Obj(
+            "Build Target" -> targetInfo.name,
```

Comment:
Minor, but in order to make these json properties more appealing for a JS client, can we use `camelCase` keys?

Reply:
Actually, what do you think about defining a case class for the entire result and define ujson codecs for them?

Reply:
Regarding the `camelCase`, totally.

Regarding the codecs approach. I guess the result we want to return the `targets` array is just the return value of `extractTargetInfo`, so we already have the case class. (Unless you mean case class for the entire return value).

I would normally be all for doing the codecs, but do you think in this situation it's useful? There's really only a few different things that the api can return right now, and it's pretty contained and straight-forward in here. I think with the html approach we tried to keep is super simple, and I tried to do the same with JSON. Again, I'm not opposed to it, but what does it really gain us here by making the codes vs just being explicit with it here?

Reply:
> Unless you mean case class for the entire return value

Yep, I also meant that.

> making the codes vs just being explicit 

To be clear: the codec can be explicit (I don't meant to auto-derive it or anything) and live in the companion object of the respective case classes. My main concern is that now that we have different formats, keeping the encoding logic together with the data logic will make the doctor implementation a bit hard to read.